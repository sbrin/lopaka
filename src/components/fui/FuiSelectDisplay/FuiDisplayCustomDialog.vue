<script lang="ts" setup>
import {ref, toRefs} from 'vue';
import {useSession} from '../../../core/session';
import { Point } from "../../../core/point";

const session = useSession();
const {setDisplay, saveDisplayCustom} = session;
const {display} = toRefs(session.state);

const emit = defineEmits(['cancelPopup', 'setCustomDisplay']);

const customWidth = ref(display.value.x);
const customHeight = ref(display.value.y);

function cancelPopup() {
    emit('cancelPopup');
}

function setCustomDisplay() {
    emit('setCustomDisplay')
    const displayCustom = new Point(customWidth.value, customHeight.value);
    setDisplay(displayCustom, true);
    saveDisplayCustom(true);
}

</script>
<template>
    <div class="fui-display-custom-dialog">
        <div class="fui-form-row fui-form-header">Custom display size</div>
        <div class="fui-form-row">
            <label class="fui-form-label fui-form-column" for="displayCustomWidth">Width:
                <input class="fui-form-input fui-form-input__size"  type="number" v-model="customWidth" id="displayCustomWidth"/>
            </label>
            <label class="fui-form-label fui-form-column"  for="displayCustomWidth">Height:
                <input class="fui-form-input fui-form-input__size"  type="number" v-model="customHeight" id="displayCustomHeight"/>
            </label>
        </div>
        <div class="buttons-bottom">
            <button class="button button_danger fui-display-custom-dialog__cancel" @click="cancelPopup">Cancel</button>
            <button class="button fui-display-custom-dialog__save" @click="setCustomDisplay">Save</button>
        </div>
    </div>
</template>
<style lang="css">
.fui-form-input__size {
    width: 68px;
}
</style>