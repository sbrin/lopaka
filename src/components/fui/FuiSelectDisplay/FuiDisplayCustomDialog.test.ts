import {mount} from '@vue/test-utils';
import {describe, expect, it} from 'vitest';
import FuiDisplayCustomDialog from './FuiDisplayCustomDialog.vue';

describe('FuiDisplayCustomDialog', () => {
    it('renders a popup element', () => {
        const wrapper = mount(FuiDisplayCustomDialog, {
            customWidth: 123,
            customHeight: 123,
        });
        expect(wrapper.find('.fui-display-custom-dialog').exists()).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
    });


    it('emits a cancelPopup event when clicked cancel', async () => {
        const wrapper = mount(FuiDisplayCustomDialog);
        const cancelButton = wrapper.find('.fui-display-custom-dialog__cancel');
        expect(cancelButton.exists()).toBe(true);

        await cancelButton.trigger('click');
        expect(wrapper.emitted('cancelPopup')).toBeTruthy();
    });

    it('emits a setCustomDisplay event when clicked save', async () => {
        const wrapper = mount(FuiDisplayCustomDialog);
        const saveButton = wrapper.find('.fui-display-custom-dialog__save');
        expect(saveButton.exists()).toBe(true);

        await saveButton.trigger('click');
        expect(wrapper.emitted('setCustomDisplay')).toBeTruthy();
    });
});
