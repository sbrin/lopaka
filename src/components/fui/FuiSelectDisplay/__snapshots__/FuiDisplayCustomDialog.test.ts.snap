// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FuiDisplayCustomDialog > renders a popup element 1`] = `
<div
  class="fui-display-custom-dialog"
>
  <div
    class="fui-form-row fui-form-header"
  >
    Custom display size
  </div>
  <div
    class="fui-form-row"
  >
    <label
      class="fui-form-label fui-form-column"
      for="displayCustomWidth"
    >
      Width: 
      <input
        class="fui-form-input fui-form-input__size"
        id="displayCustomWidth"
        type="number"
      />
    </label>
    <label
      class="fui-form-label fui-form-column"
      for="displayCustomWidth"
    >
      Height: 
      <input
        class="fui-form-input fui-form-input__size"
        id="displayCustomHeight"
        type="number"
      />
    </label>
  </div>
  <div
    class="buttons-bottom"
  >
    <button
      class="button button_danger fui-display-custom-dialog__cancel"
    >
      Cancel
    </button>
    <button
      class="button fui-display-custom-dialog__save"
    >
      Save
    </button>
  </div>
</div>
`;
